packages/serve/startDevServer.ts

```diff
@@ -0,0 +1,40 @@
+import * as Server from "webpack-dev-server/lib/Server";
+
+/**
+ *
+ * Starts the devServer
+ *
+ * @param {Object} compiler - a webpack compiler
+ * @param {Object} options - devServer options
+ * @param {Function} onListening - optional callback for when the server starts listening
+ * 
+ * @returns {Void} 
+ */
+export default function startDevServer(compiler, options, onListening): void {
+    const firstWpOpt = compiler.compilers
+        ? compiler.compilers[0].options
+        : compiler.options;
+    const devServerOptions = firstWpOpt.devServer || {};
+
+    const host = options.host || devServerOptions.host || 'localhost';
+    const port = options.port || devServerOptions.port || 8080;
+    // socket should not have a default value, because it should only be used if the
+    // user explicitly provides it
+    const socket = options.socket || devServerOptions.socket;
+
+    options.host = host;
+    options.port = port;
+    if (socket) {
+        options.socket = socket;
+    }
+
+    const server = new Server(compiler, options);
+    server.listen(socket || port, host, (err): void => {
+        if (err) {
+            throw err;
+        }
+        if (typeof onListening === 'function') {
+            onListening();
+        }
```

Comment:
We should solve this on `webpack-dev-server` side, not here

Reply:
> We should solve this on `webpack-dev-server` side, not here

This `onListening` is different, it is just a callback as one of the function parameters. It can be removed if needed.